File: OCH_Tao/ABC/B/283.py
N = int(input())
A = list(map(int,input().split()))
Q = int(input())
for i in range(Q):
  n,*k = map(int,input().split())
  if n==1:
    A[k[0]-1]=k[1]
  else:
    print(A[k[0]-1])

File: OCH_Tao/ABC/B/283.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
input.shift();
let A = input.shift().split(" ").map(x => parseInt(x, 10));
const Q = parseInt(input.shift(), 10);
for (let i = 0; i < Q; i++) {
  const [n, ...k] = input[i].split(" ").map(x => parseInt(x, 10));
  if (n === 1) {
    A[k[0] - 1] = k[1];
  } else {
    console.log(A[k[0] - 1]);
  }
}